sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-pre-validations-processor.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { SystemUsersService } from "@sims/services";
+import { CASSupplier, SupplierStatus } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import {
+  CASEvaluationResult,
+  CASEvaluationStatus,
+} from "../cas-supplier.models";
+import { Repository } from "typeorm";
+import { CASEvaluationResultProcessor, ProcessorResult } from ".";
+import { CASAuthDetails } from "@sims/integrations/cas";
+
+/**
+ * Assert the student can be added to CAS.
+ */
+@Injectable()
+export class CASPreValidationsProcessor extends CASEvaluationResultProcessor {
+  constructor(
+    private readonly systemUsersService: SystemUsersService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    super();
+  }
+
+  /**
+   * Process the result of a pre-validation failed, setting
+   * the student CAS supplier for manual intervention.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param _auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  async process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    _auth: CASAuthDetails,
```

Comment:
If we are pursuing the auth, shouldn't it not be optional as signature?

Reply:
It does not matter if optional or not because the consumers must provide all parameters either way for all the processors.
If it is to make it clear to the developer looking into the processor, the "_" will achieve the same.